sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/BuilderHelperTests.java

```diff
@@ -0,0 +1,568 @@
+     * pipeline.
+     */
+    @ParameterizedTest
+    @MethodSource("customApplicationIdInUAStringSupplier")
+    void shareClientCustomApplicationIdInUAString(List<String> applicationIds) throws MalformedURLException {
+        String logOptionsUA = applicationIds.get(0);
+        String clientOptionsUA = applicationIds.get(1);
+        String expectedUA = applicationIds.get(2);
+        ShareClient shareClient = new ShareClientBuilder()
+            .endpoint(ENDPOINT)
+            .shareName("share")
+            .credential(CREDENTIALS)
+            .httpLogOptions(new HttpLogOptions().setApplicationId(logOptionsUA))
+            .clientOptions(new ClientOptions().setApplicationId(clientOptionsUA))
+            .httpClient(new ApplicationIdUAStringTestClient(expectedUA))
+            .buildClient();
+
+        shareClient.getHttpPipeline().send(request(shareClient.getShareUrl()))
+            .as(StepVerifier::create)
+            .assertNext(response -> {
+                assertEquals(200, response.getStatusCode());
+            })
+            .verifyComplete();
+    }
+
+    /**
+     * Tests that a user application id will be honored in the UA string when using the file client builder's default
+     * pipeline.
+     */
+    @ParameterizedTest
+    @MethodSource("customApplicationIdInUAStringSupplier")
+    void fileClientCustomApplicationIdInUAString(List<String> applicationIds) throws MalformedURLException {
+        String logOptionsUA = applicationIds.get(0);
+        String clientOptionsUA = applicationIds.get(1);
+        String expectedUA = applicationIds.get(2);
+        ShareFileClientBuilder fileClientBuilder = new ShareFileClientBuilder()
+            .endpoint(ENDPOINT)
+            .shareName("fileSystem")
+            .resourcePath("path")
+            .credential(CREDENTIALS)
+            .httpLogOptions(new HttpLogOptions().setApplicationId(logOptionsUA))
+            .clientOptions(new ClientOptions().setApplicationId(clientOptionsUA))
+            .httpClient(new ApplicationIdUAStringTestClient(expectedUA));
+
+        ShareDirectoryClient directoryClient = fileClientBuilder.buildDirectoryClient();
+        StepVerifier.create(directoryClient.getHttpPipeline().send(request(directoryClient.getDirectoryUrl())))
+            .assertNext(response -> {
+                assertEquals(200, response.getStatusCode());
+            })
+            .verifyComplete();
+
+        ShareFileClient fileClient = fileClientBuilder.buildFileClient();
+        StepVerifier.create(fileClient.getHttpPipeline().send(request(fileClient.getFileUrl())))
+            .assertNext(response -> {
+                assertEquals(200, response.getStatusCode());
+            })
+            .verifyComplete();
+    }
+
+    private static Stream<List<String>> customApplicationIdInUAStringSupplier() {
+        return Stream.of(
+            new ArrayList<String>() {
```

Comment:
For parameterized tests with multiple arguments in the parameterized test I'd recommend using `Arguments.of` instead of a list of values. So, the first test case here would be `Arguments.of("log-options-id", null, "log-options-id")`. Then for the test using these parametarizations you can change the signature to use a non-list parameter, so in this case it'd be `void fileClientCustomApplicationIdInUAString(String logOptionsUA, String clientOptionsUA, String expectedUA)`, which then cleans up the test code to no longer require getting certain values from a list or requiring an unchecked cast. JUnit will handle type coercion from `Arguments` to the method parameters.